src/request.rs

```diff
@@ -79,6 +85,8 @@ pub struct RequestBuilder<B = [u8; 0]> {
     accept_invalid_certs: bool,
     #[cfg(feature = "tls")]
     accept_invalid_hostnames: bool,
+    #[cfg(feature = "tls-rustls")]
+    client_config: SkipDebug<Option<Arc<ClientConfig>>>,
```

Comment:
This differs slightly in the approach compared to the existing `accept_invalid_*` fields but is strictly more general as it supports e.g. setting up client authentication. I could also reuse the existing fields marking them as `#[cfg(any(feature = "tls", feature  = "tls-rustls"))]` and implementing their functionality in terms of `rustls::ClientConfig` if you prefer.

Another option I see here would be to align this by replacing the `accept_invalid_*` fields by a `tls_connector: Option<TlsConnector>` which would also allow full access to the `native-tls` configuration but would imply an incompatible API change.

Reply:
Going for `tls_connector: Option<TlsConnector>` would also allow us to close #41 so this does look like a nice choice.